src/hnsw/htab_cache.h

```diff
@@ -0,0 +1,27 @@
+#ifndef LDB_HNSW_HTAB_CACHE_H
+#define LDB_HNSW_HTAB_CACHE_H
+#include "postgres.h"
+
+#include <utils/hsearch.h>
+#include <utils/memutils.h>
+
+typedef int HTABCacheKey;
+typedef struct HTABCacheEntry
+{
+    HTABCacheKey key;
+    void        *value;
+} HTABCacheEntry;
+
+typedef struct HTABCache
+{
+    HTAB   *htab;
+    HASHCTL hctl;
+} HTABCache;
+
+HTABCache cache_create(char *name, MemoryContext ctx);
+bool      cache_remove(HTABCache *cache, HTABCacheKey *key);
```

Comment:
Can you document the assumptions of this ADT in this file?

E.g. currently the caller creates the memory context, passes ownership to this ADT and this ADT destroys it upon `cache_destroy`. 
Also answer questions like: 
Who owns *key and *entry pointers passed into these API functions? Does the ADT make assumptions about how long those pointers live? (if I understand it correctly, the ADT assumes that anything passed in lives longer than the cache ADT itself. This invariant must be followed by the user of the ADT so this interface file is a good place to document it)

Reply:
Added comments [here](https://github.com/lanterndata/lantern/pull/160/files#diff-19338c152ee6b524d410f905b9fe9781d838b4f8b9db6f5d964c609824e31d1eR10-R21)